website/internal/views/__init__.py

```diff
@@ -48,5 +73,24 @@ def themes():
 
 @internal_site.route("/graph")
 def graph():
+    return render_template("internal/graph.html")
+
+
+@internal_site.route('/graph/data')
+def graph_data():
+    si = StringIO()
+    cw = csv.writer(si)
     data = data_interface.get_user_graph_data(utilities.session.get_active_user()['user_id'])
-    return render_template("internal/graph.html", data=data)
+    cw.writerow(["date", "close"])
+    keys = sorted(data.keys())
+    for k in keys:
+        cw.writerow([k, data[k]])
+    output = make_response(si.getvalue())
+    output.headers["Content-Disposition"] = "attachment; filename=export.csv"
+    output.headers["Content-type"] = "text/csv"
+    return output
+
+
+@internal_site.route('/graph_test')
```

Comment:
As you removed the graph_test.html file, shouldn't we remove this endpoint as well?